src/detectors/FEMC/FEMC.cc

```diff
@@ -18,41 +24,99 @@
 #include "factories/calorimetry/CalorimeterIslandCluster_factory.h"
 #include "factories/calorimetry/CalorimeterTruthClustering_factory.h"
 #include "factories/calorimetry/TrackClusterMergeSplitter_factory.h"
+#include "services/geometry/dd4hep/DD4hep_service.h"
+#include "services/log/Log_service.h"
 
 extern "C" {
 void InitPlugin(JApplication* app) {
 
   using namespace eicrecon;
 
   InitJANAPlugin(app);
+
+  auto log_service = app->GetService<Log_service>();
+  auto mLog        = log_service->logger("FEMC");
+
   // Make sure digi and reco use the same value
   decltype(CalorimeterHitDigiConfig::capADC) EcalEndcapP_capADC =
       16384; //16384, assuming 14 bits. For approximate HGCROC resolution use 65536
-  decltype(CalorimeterHitDigiConfig::dyRangeADC) EcalEndcapP_dyRangeADC   = 3 * dd4hep::GeV;
+  decltype(CalorimeterHitDigiConfig::dyRangeADC) EcalEndcapP_dyRangeADC   = 100 * dd4hep::GeV;
   decltype(CalorimeterHitDigiConfig::pedMeanADC) EcalEndcapP_pedMeanADC   = 200;
   decltype(CalorimeterHitDigiConfig::pedSigmaADC) EcalEndcapP_pedSigmaADC = 2.4576;
   decltype(CalorimeterHitDigiConfig::resolutionTDC) EcalEndcapP_resolutionTDC =
       10 * dd4hep::picosecond;
-  app->Add(new JOmniFactoryGeneratorT<CalorimeterHitDigi_factory>(
-      "EcalEndcapPRawHits", {"EventHeader", "EcalEndcapPHits"},
-      {"EcalEndcapPRawHits", "EcalEndcapPRawHitAssociations"},
-      {
-          .eRes      = {0.11333 * sqrt(dd4hep::GeV), 0.03,
-                        0.0 * dd4hep::GeV}, // (11.333% / sqrt(E)) \oplus 3%
-          .tRes      = 0.0,
-          .threshold = 0.0,
-          // .threshold = 15 * dd4hep::MeV for a single tower, applied on ADC level
-          .capADC        = EcalEndcapP_capADC,
-          .capTime       = 100, // given in ns, 4 samples in HGCROC
-          .dyRangeADC    = EcalEndcapP_dyRangeADC,
-          .pedMeanADC    = EcalEndcapP_pedMeanADC,
-          .pedSigmaADC   = EcalEndcapP_pedSigmaADC,
-          .resolutionTDC = EcalEndcapP_resolutionTDC,
-          .corrMeanScale = "0.03",
-          .readout       = "EcalEndcapPHits",
-      },
-      app // TODO: Remove me once fixed
-      ));
+  const double EcalEndcapP_sampFrac = 0.029043; // updated with ratio to ScFi model
+  decltype(CalorimeterHitDigiConfig::corrMeanScale) EcalEndcapP_corrMeanScale =
+      fmt::format("{}", 1.0 / EcalEndcapP_sampFrac);
+  const double EcalEndcapP_nPhotonPerGeV          = 1500;
+  const double EcalEndcapP_PhotonCollectionEff    = 0.5;
+  const unsigned long long EcalEndcapP_totalPixel = 4 * 159565ULL;
+
+  int FEMCHomoScFi = 0;
```

Comment:
FYI, "homo" is a slur in many languages. We should use "homogeneous".
```suggestion
  int FEMCHomogeneousScFi = 0;
```